src/Chapter3.hs

```diff
@@ -907,9 +998,23 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append (Gold x) (Gold y) = Gold (x + y)
+
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append x y = x ++ y
+
+instance Append (Maybe [a]) where
+    append :: Maybe [a] -> Maybe [a] -> Maybe [a]
+    append (Just x) (Just y) =  Just (x ++ y)
```

Comment:
This function is partial, because it doesn't handle cases when you pass `Nothing` to it